### PR TITLE
Add initial caching

### PR DESCRIPTION
**Pull request by serverwentdown**
If it's an okay feature to add, might consider replicating the same features in [`flutter-action`](https://github.com/subosito/flutter-action)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,6 +174,15 @@ jobs:
         run: pub run test
 ```
 
+## Cached example
+
+```yml
+      - uses: dart-lang/setup-dart@v1.3
+        with:
+          sdk: '2.9.0'
+          cache: true # Should only be used when a fixed SDK version is specified
```
Comment (devoncarew) [inline README.md:183]: Why only with a fixed version of the SDK? Otherwise, we may cache and use version x, even when x+1 becomes available?

**Comment by devoncarew** [comment]
Thanks for the contribution! I have one comment in-line in the PR, and a question. What kind of perf benefit would this get - caching the SDK? I'm trying to balance that vs keeping this action simple.

**Comment by devoncarew** [comment]
I wonder if this PR wouldn't be necessary if we instead addressed issue https://github.com/dart-lang/setup-dart/issues/2?

File: README.md (hunk 1 above)
Comment (serverwentdown) [reply README.md:183]: Yes

**Comment by natebosch** [comment]
I think it would be better to use etags which should let us cache unconditionally instead of only for fixed version SDKs.

**Comment by serverwentdown** [comment]
About 2s is shaved off the download and decompression on Ubuntu GitHub Action hosted runners. On self-hosted runners, this can potentially save time depending on connectivity to GitHub and Google.
https://github.com/heacare/setup-dart-test/runs/7764490992?check_suite_focus=true

I was using caching on [`flutter-action`](https://github.com/subosito/flutter-action) and it saved substantial setup time.

**Comment by mit-mit** [comment]
@devoncarew is this still relevant in the new reimplementation of this action?

**Pull request closed without merging**